File: src/main/java/spring/webmvc/presentation/dto/response/MemberResponse.java
package spring.webmvc.presentation.dto.response;

import java.time.Instant;
import java.time.LocalDate;

import spring.webmvc.domain.model.entity.Member;

public record MemberResponse(
	Long id,
	String account,
	String name,
	String phone,
	LocalDate birthDate,
	Instant createdAt
) {
	public MemberResponse(Member member) {
		this(
			member.getId(),
			member.getAccount(),
			member.getName(),
			member.getPhone(),
			member.getBirthDate(),
			member.getCreatedAt()
		);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/response/MemberResponse.kt
package spring.web.kotlin.presentation.dto.response

import spring.web.kotlin.domain.model.entity.Member
import java.time.Instant
import java.time.LocalDate

data class MemberResponse(
    val id: Long,
    val account: String,
    val name: String,
    val phone: String,
    val birthDate: LocalDate,
    val createdAt: Instant,
) {
    constructor(member: Member) : this(
        id = checkNotNull(member.id),
        account = member.account,
        name = member.name,
        phone = member.phone,
        birthDate = member.birthDate,
        createdAt = member.createdAt,
    )
}
